src/raglite/_database.py

```diff
@@ -44,6 +46,23 @@
 )
 
 
+def _get_metadata_column() -> type[TypeDecorator[Any]]:
+    """Get the appropriate metadata column type based on database dialect."""
+    # We can't determine dialect at import time, so we'll use a callable
+    # that gets resolved when the table is created
+
+    class MetadataJSON(TypeDecorator[Any]):
+        impl = JSON
+        cache_ok = True
+
+        def load_dialect_impl(self, dialect: Any) -> Any:
+            if dialect.name == "postgresql":
+                return dialect.type_descriptor(JSONB())
+            return dialect.type_descriptor(JSON())
+
+    return MetadataJSON
```

Comment:
```suggestion
def _get_metadata_column() -> type[TypeDecorator[Any]]:
    """Get the appropriate metadata column type based on database dialect."""
    # We can't determine dialect at import time, so we'll use a callable
    # that gets resolved when the table is created

    class MetadataJSON(TypeDecorator[Any]):
        impl = JSON
        cache_ok = True

        def load_dialect_impl(self, dialect: Any) -> Any:
            if dialect.name == "postgresql":
                return dialect.type_descriptor(JSONB())
            return dialect.type_descriptor(JSON())

    return MetadataJSON
```

For my information, can we use JSON for Postgres too, or are there downsides to doing that that we want to avoid?

Since this function doesn't return a column but a type (or actually a type factory), I would rename it to: `_metadata_column_type`.